.github/workflows/ci-matrix.yaml.njk

```diff
@@ -0,0 +1,53 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: {{name}}
+on:
+  push:
+    branches:
+    - main
+    paths:
+    - '{{path}}/**'
+    - '.github/workflows/{{name}}.yaml'
+  pull_request:
+    paths:
+    - '{{path}}/**'
+    - '.github/workflows/{{name}}.yaml'
+  pull_request_target:
+    types: [labeled]
+    paths:
+    - '{{path}}/**'
+    - '.github/workflows/{{name}}.yaml'
+  schedule:
+  - cron:  '0 0 * * 0'
+jobs:
+  test:
+    if: github.event.action != 'labeled' || github.event.label.name == 'actions:force-run'
+    strategy:
+      matrix:
+        path:
+          # TODO: Replace these examples below 
+          - 'example/subfolder/1'
+          - 'example/subfolder/2'
+    uses: ./.github/workflows/test.yaml
+    with:
+      name: '{{name}}'
+      path: '${% raw %}{{ matrix.path }}{% endraw %}'
+  remove_label:
+    if: github.event.action == 'labeled' && github.event.label.name == 'actions:force-run' && always()
+    uses: ./.github/workflows/remove-label.yaml
+  flakybot:
+    if: github.event_name == 'schedule' && always()
```

Comment:
Might be worth adding a comment about the use of `always()`, I had to read the docs.